### ISSUE TITLE
Upon container creation, apache configuration files in mounted volumes are overwritten

### ISSUE THREAD

**Issue by fseidl** [bitnami/bitnami-docker-dokuwiki#62]
**Description**

This might be a problem of bitnami-docker-apache, but I'm experiencing it with bitnami-docker-dokuwiki.

The documentation says (under "Notable Changes"): 

> ... By default, and if the configuration volume does not exist, the configuration files will be regenerated each time the container is created. Users wanting to apply custom Apache configuration files are advised to mount a volume for the configuration at /opt/bitnami/apache/conf, or mount specific configuration files individually.

I have tried both: To mount a volume at /opt/bitnami/apache/conf and to mount the configuration file vhosts/dokuwiki-vhost.conf individually.

In either case, the file does not survive container creation. It gets changed back to factory defaults. (This does not happen for /opt/bitnami/php/conf or e.g. php.ini.)

I'm using docker-compose.

Declaring the mount read-only results in the error

> Error executing 'postInstallation': EROFS: read-only file system, open '/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf'

(A mount for php.ini can be declared read-only just fine.)

**Steps to reproduce the issue:**
1. Create a dokuwiki service using docker-compose with a volume for dokuwiki-vhost.conf. 
2. Change the contents of dokuwiki-vhosts.conf to something other than the default.
3. docker-compose down ; docker-compose up
4. Notice that the changes from 2. are gone.

The `docker-compose.yml`
```
version: '3.5'

networks:
  default:
    external:
      name: theweb
volumes:
  dokuwiki_data:
    driver: local

services:
  dokuwiki:
    image: 'bitnami/dokuwiki:latest'
    ports:
      - 82:80
    labels:
        - "traefik.docker.network=theweb"
        - "traefik.enable=true"
        - "traefik.frontend.rule=Host:wiki.example.com"
        - "traefik.port=80"
        - "traefik.protocol=http"
        - "traefik.frontend.passHostHeader=true"
    volumes:
      - 'dokuwiki_data:/bitnami'
      - './dokuwiki-vhost.conf:/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf:ro'
      - './php.ini:/opt/bitnami/php/conf/php.ini:ro'
```

**Describe the results you received:**

Upon container creation or recreation, apache configuration files are regenerated despite being mounted.

**Describe the results you expected:**

Configuration files supplied from outside to be left alone upon container recreation.

**Version**

- Output of `docker version`:

```
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.1
 Git commit:        2d0083d
 Built:             Wed Jul  3 12:13:59 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.1
  Git commit:       2d0083d
  Built:            Mon Jul  1 19:31:12 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

- Output of `docker info`:

```
Containers: 16                                                                                                                         
 Running: 12                                                                                                                           
 Paused: 0                                                                                                                             
 Stopped: 4                                                                                                                            
Images: 45                                                                                                                             
Server Version: 18.09.7                                                                                                                
Storage Driver: btrfs                                                                                                                  
 Build Version: Btrfs v4.15.1                                                                                                          
 Library Version: 102                                                                                                                  
Logging Driver: json-file                                                                                                              
Cgroup Driver: cgroupfs                                                                                                                
Plugins:                                                                                                                               
 Volume: local                                                                                                                         
 Network: bridge host macvlan null overlay                                                                                             
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 
runc version: N/A
init version: v0.18.0 (expected: fec3683b971d9c3ef73f284f176672c44b448662)
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-55-generic 
Operating System: Ubuntu 18.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 15.66GiB
Name: docker
ID: AKFO:FCQB:NF6R:NVJF:FRIU:LMJD:WXB4:OJR5:56BD:GNMT:N62J:IALQ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.15+
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Docker host is a KVM virtual machine.

**Comment by andresbono** [comment]
Hi @fseidl, thanks for reporting this issue, it seems this is not working properly. We will get back to you when we have more news.

**Comment by carlosbustillordguez** [comment]
I have the same problem in Kubernetes when I mount a volume from a configmap: 
`Error executing 'postInstallation': EROFS: read-only file system, open '/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf'`

Also, I tried to mount as volume the file `/opt/bitnami/apache/conf/vhosts/htaccess/dokuwiki-htaccess.conf`, but I got the same error:

`Error executing 'postInstallation': EROFS: read-only file system, open '/opt/bitnami/apache/conf/vhosts/htaccess/dokuwiki-htaccess.conf'`

I noted if the file you want to mount doesn't belong to the default files, it works.

**Comment by andresbono** [comment]
Hi @carlosbustillordguez, thanks for letting us know. We didn't address the issue yet.

In the meantime, is your deployment running properly by using other files different than the default ones?

**Comment by carlosbustillordguez** [comment]
Hi @andresbono, thanks for your reply!! 

My main issue was when I tried to enable [clean URLs for Dokuwiki](https://www.dokuwiki.org/rewrite) using the htaccess method, for that I modified the file `/opt/bitnami/apache/conf/vhosts/htaccess/dokuwiki-htaccess.conf` using a configmap but doesn't works due the error above... so I added a new file using a configmap for instance: `/opt/bitnami/apache/conf/vhosts/htaccess/dokuwiki-htaccess-clean-urls.conf` and it worked. 

Then to include this file in the Apache config, I modified the with a configmap the file ''/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf'', but I got the error:

`'Error executing 'postInstallation': EROFS: read-only file system, open '/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf'`

Finally, to enable the clean URLs in Dokuwiki I used the second option (Rewriting is done by DokuWiki). In this case, I don't need to modify files outside `/bitname/dokuwiki/` directory.

So, if you add new files different as the default ones the deployment works.

Regards, Carlos

**Comment by andresbono** [comment]
Understood, thanks for this extra information.

**Comment by marcosbc** [comment]
Hi, we recently released DokuWiki `20180422.4.0-debian-10-r0` which now allows Apache configuration files to be mounted/configurable, meaning this issue should be fixed. See: https://github.com/bitnami/bitnami-docker-dokuwiki#2018042240-debian-10-r0 and https://github.com/bitnami/bitnami-docker-dokuwiki#customize-this-image

P.S.: Let's continue the clean URLs on https://github.com/bitnami/bitnami-docker-dokuwiki/issues/52

I'm closing this issue, but please let us know if you think this is not yet solved.
